libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/ServerPickerActivity.java

```diff
@@ -244,4 +243,27 @@ public void rebuildDisplay() {
             }
         }
     }
+
+    private class AuthConfigTask extends AsyncTask<Void, Void, Void> {
+
+        @Override
+        protected Void doInBackground(Void... nothings) {
+            final String loginServer = SalesforceSDKManager.getInstance().getLoginServerManager().getSelectedLoginServer().url;
```

Comment:
Simple `AsyncTask` that runs the query in a background thread and posts the result on the main thread when done and dismissed itself.